darwin/client.py

```diff
@@ -130,6 +132,9 @@ def put(
             time.sleep(10)
             return self.put(endpoint, payload=payload, retry=False)
 
+        if response.status_code != 200 and block_unsuccessful_requests:
+            raise UnsuccessfulResponse()
```

Comment:
Could we add the response body or more information to the error?
Logging something could also be useful.